packages/headless/src/features/generated-answer/generated-answer-request.test.ts

```diff
@@ -0,0 +1,150 @@
+    const queryParams = constructAnswerQueryParams(
+      streamAnswerAPIStateMockWithATabWithAnExpression,
+      buildMockNavigatorContextProvider()()
+    );
+
+    expect(queryParams).toEqual(
+      expectedStreamAnswerAPIParamWithATabWithAnExpression
+    );
+  });
+
+  it('should not include tab info when there is NO tab', () => {
+    const queryParams = constructAnswerQueryParams(
+      streamAnswerAPIStateMockWithoutAnyTab,
+      buildMockNavigatorContextProvider()()
+    );
+
+    expect(queryParams).toEqual(expectedStreamAnswerAPIParamWithoutAnyTab);
+  });
+
+  it('should merge filter expressions in request constant query when expression is selected', () => {
+    const queryParams = constructAnswerQueryParams(
+      streamAnswerAPIStateMockWithStaticFiltersSelected,
+      buildMockNavigatorContextProvider()()
+    );
+
+    expect(queryParams).toEqual(
+      expectedStreamAnswerAPIParamWithStaticFiltersSelected
+    );
+  });
+
+  it('should not include filter info when there is NO filter', () => {
+    const queryParams = constructAnswerQueryParams(
+      streamAnswerAPIStateMockWithoutAnyFilters,
+      buildMockNavigatorContextProvider()()
+    );
+    expect(queryParams).toEqual(expectedStreamAnswerAPIParam);
+  });
+
+  it('should not include non-selected filters and empty filters', () => {
+    const queryParams = constructAnswerQueryParams(
+      streamAnswerAPIStateMockWithNonValidFilters,
+      buildMockNavigatorContextProvider()()
+    );
+    expect(queryParams).toEqual(expectedStreamAnswerAPIParam);
+  });
+
+  it('should merge multiple filter expressions and a tab expression', () => {
+    const queryParams = constructAnswerQueryParams(
+      streamAnswerAPIStateMockWithStaticFiltersAndTabExpression,
+      buildMockNavigatorContextProvider()()
+    );
+    expect(queryParams).toEqual(
+      expectedStreamAnswerAPIParamWithStaticFiltersAndTabExpression
+    );
+  });
+
+  it('should not include advanced search queries when there are no advanced search queries', () => {
+    const queryParams = constructAnswerQueryParams(
+      streamAnswerAPIStateMockWithStaticFiltersAndTabExpressionWithEmptyCQ,
+      buildMockNavigatorContextProvider()()
+    );
+    expect(queryParams).toEqual(
+      expectedStreamAnswerAPIParamWithStaticFiltersAndTabExpressionWithoutAdvancedCQ
+    );
+  });
+
+  it('should accept an undefined SearchAction', () => {
+    const queryParams = constructAnswerQueryParams(
+      streamAnswerAPIStateMockWithoutSearchAction,
+      buildMockNavigatorContextProvider()()
+    );
+
+    expect(queryParams).toEqual(
+      expectedStreamAnswerAPIParamWithoutSearchAction
+    );
+  });
+
+  it('should include all analytics fields when usage is fetch', () => {
+    const queryParams = constructAnswerQueryParams(
+      streamAnswerAPIStateMock,
+      buildMockNavigatorContextProvider()()
+    );
+
+    // Verify that all analytics fields are present including volatile ones
```

Comment:
What does volatile mean in this context?

Reply:
this is not something I added, but danny was saying time related stuff in the analytics